speckle/Cache.py

```diff
@@ -0,0 +1,294 @@
+
+        # Sanitize
+        host = host.strip("/")
+        host = host.strip("\\")
+        email = email.strip()
+
+        conn = self.try_connect()
+        if conn == None:
+            raise Exception("Failed to connect to database.")
+
+        with conn:
+            c = conn.cursor()
+            try:
+                c.execute(""" DELETE FROM Account WHERE RestApi = ? AND Email = ?""", (host, email,))
+                conn.commit()
+            except sqlite3.IntegrityError as e:
+                self.log("Account already exists in database.")
+
+    def delete_all(self, table="CachedObject"):
+        """Deletes all objects from table in database
+
+        Deletes all objects from specified table in the database specified by self.db_path.
+        
+        Keyword Arguments:
+            table {str} -- Table in database to delete all entries from {default: {"CachedObject"}}
+        """
+
+        conn = self.try_connect()
+        if conn == None:
+            raise Exception("Failed to connect to database.")
+
+        with conn:
+            c = conn.cursor()
+            try:
+                c.execute(" DELETE FROM " + table)
+                conn.commit()
+            except sqlite3.IntegrityError as e:
+                self.log("Failed to clear table{}.".format(table))
+
+    def get_all_accounts(self):
+        """Get all accounts in database
+
+        Gets all user profiles from the database specified by self.db_path.
+
+        Returns:
+            Accounts -- A list of tuples containing user profile data        
+
+        """
+
+        conn = self.try_connect()
+        if conn == None:
+            raise Exception("Failed to connect to database.")
+
+        with conn:
+            c = conn.cursor()
+            try:
+                c.execute(""" SELECT * FROM Account""")
+                rows = c.fetchall()
+                return rows
+
+            except sqlite3.IntegrityError as e:
+                raise
+
+    def get_account(self, host, email):
+        """Gets user profile from database
+
+        Gets a user profile from the database specified by self.db_path.
+        
+        Arguments:
+            host {str} -- Speckle server
+            email {str} -- User email address
+
+        Returns:
+            Account -- A tuple containing user profile data
+        """        
+        conn = self.try_connect()
+        if conn == None:
+            self.log("Failed to access database.")
+            return False
+
+        with conn:
+            c = conn.cursor()
+            try:
+                c.execute(""" SELECT * FROM Account WHERE RestApi = ? AND Email = ?""", (host, email,))
+                #c.execute(""" SELECT * FROM Account WHERE Email=?""", (email,))
+                rows = c.fetchall()
+                if len(rows) > 0:
+                    return rows[0]
+                return None
+
+            except sqlite3.IntegrityError as e:
+                raise
+
+
+if __name__ == "__main__":
```

Comment:
Why is there a __main__ method here? Are we likely to run this code directly from file? Also, why are we creating dummy accounts by default?

Reply:
**main()** is just for testing, when `Cache.py` is executed by itself. Can be removed, no biggie.

Reply:
If it's just for testing I reckon we should dump the code in the `tests` folder in a file called `test_cache.py`. Then you can use standard `pytest` to write your tests and make sure it all works. To be honest I'm not sure what all this local db cache is for so I trust you to write the appropriate tests for it 😉